homeassistant/components/vasttrafik/config_flow.py

```diff
@@ -0,0 +1,350 @@
+                    CONF_LINES: departure.get(CONF_LINES, []),
+                    CONF_TRACKS: departure.get(CONF_TRACKS, []),
+                    CONF_DELAY: departure.get(CONF_DELAY, DEFAULT_DELAY),
+                },
+                unique_id=None,
+            )
+            for departure in import_data.get(CONF_DEPARTURES, [])
+        ]
+
+        return self.async_create_entry(
+            title="Västtrafik",
+            data={
+                CONF_KEY: import_data[CONF_KEY],
+                CONF_SECRET: import_data[CONF_SECRET],
+            },
+            subentries=subentries,
+        )
+
+
+class VasttrafikSubentryFlow(ConfigSubentryFlow):
+    """Handle a subentry config flow for Västtrafik departure boards."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the subentry flow."""
+        super().__init__()
+        self._search_results: list[dict] = []
+        self._selected_station: str = ""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> SubentryFlowResult:
+        """Handle the initial step of adding a departure board."""
+        if user_input is not None:
+            search_query = user_input.get("search_query", "").strip()
+            if not search_query:
+                errors = {"base": "search_required"}
+            elif len(search_query) < 2:
+                errors = {"base": "search_too_short"}
+            else:
+                parent_entry = self._get_entry()
+
+                stations, error_code = await search_stations(
+                    self.hass, parent_entry, search_query
+                )
+                if error_code:
+                    errors = {"base": error_code}
+                elif not stations:
+                    errors = {"base": "no_stations_found"}
+                else:
+                    self._search_results = stations
+                    return await self.async_step_select_station()
+        else:
+            errors = {}
+
+        search_schema = vol.Schema(
+            {
+                vol.Required("search_query"): vol.All(str, vol.Length(min=2)),
+            }
+        )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=search_schema,
+            errors=errors,
+        )
+
+    async def async_step_select_station(
+        self, user_input: dict[str, str] | None = None
+    ) -> SubentryFlowResult:
+        """Select station and configure departure sensor."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            self._selected_station = user_input["station"]
+            return await self.async_step_configure()
+
+        station_options = {
+            station["value"]: station["label"] for station in self._search_results
+        }
+
+        station_schema = vol.Schema(
+            {
+                vol.Required("station"): vol.In(station_options),
```

Comment:
If you use the `SelectSelector`instead this could be presented nicely with the proper labels instead of just the station value.